router-tests/feature_flags_test.go

```diff
@@ -0,0 +1,90 @@
+package integration
+
+import (
+	"github.com/stretchr/testify/require"
+	"github.com/wundergraph/cosmo/router-tests/testenv"
+	"testing"
+)
+
+func TestFeatureFlags(t *testing.T) {
+
+	t.Parallel()
+
+	t.Run("Base feature graph schema is served when no feature flag is enabled", func(t *testing.T) {
+
+		t.Parallel()
+
+		testenv.Run(t, &testenv.Config{}, func(t *testing.T, xEnv *testenv.Environment) {
+			res := xEnv.MakeGraphQLRequestOK(testenv.GraphQLRequest{
+				Query: `{ employees { id productCount } }`,
+			})
+			require.Empty(t, res.Response.Header.Get("X-Feature-Flag"))
+			require.JSONEq(t, `{"errors":[{"message":"field: productCount not defined on type: Employee","path":["query","employees","productCount"]}],"data":null}`, res.Body)
+		})
+	})
+
+	t.Run("Base feature graph schema is served when feature flag does not exist / header", func(t *testing.T) {
+
+		t.Parallel()
+
+		testenv.Run(t, &testenv.Config{}, func(t *testing.T, xEnv *testenv.Environment) {
+			res := xEnv.MakeGraphQLRequestOK(testenv.GraphQLRequest{
+				Query: `{ employees { id productCount } }`,
+				Header: map[string][]string{
+					"X-Feature-Flag": {"nonexistent"},
+				},
+			})
+			require.Empty(t, res.Response.Header.Get("X-Feature-Flag"))
+			require.JSONEq(t, `{"errors":[{"message":"field: productCount not defined on type: Employee","path":["query","employees","productCount"]}],"data":null}`, res.Body)
+		})
+	})
+
+	t.Run("Base feature graph schema is served when feature flag does not exist / cookie", func(t *testing.T) {
+
+		t.Parallel()
+
+		testenv.Run(t, &testenv.Config{}, func(t *testing.T, xEnv *testenv.Environment) {
+			res := xEnv.MakeGraphQLRequestOK(testenv.GraphQLRequest{
+				Query: `{ employees { id productCount } }`,
+				Header: map[string][]string{
+					"Cookie": {"feature_flag=nonexistent"},
+				},
+			})
+			require.Empty(t, res.Response.Header.Get("X-Feature-Flag"))
+			require.JSONEq(t, `{"errors":[{"message":"field: productCount not defined on type: Employee","path":["query","employees","productCount"]}],"data":null}`, res.Body)
+		})
+	})
+
+	t.Run("Should replace product feature graph when feature flag is sent over header", func(t *testing.T) {
```

Comment:
Please add 2 more tests with a Middleware (you can configure it on testenv.Config{}) for the Feature Graph, validating that we're setting "X-Feature-Flag" Header on all Subgraph requests.
It's possible to publish the same Schema and URL under a FF, but the internal implementation differs. In such a case, the Header is needed in the Subgraph to understand which FF is being used.

Reply:
This has been implemented yesterday. It is also validated in the tests.

Reply:
I have created an internal ticket.